### PR TITLE
Symex max depth default max

### PR DESCRIPTION
**Pull request by TGWDB**
This PR implements the recommended fix in [#2295](https://github.com/diffblue/cbmc/issues/2295).

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by martin-cs — APPROVED** [review]

**Comment by kroening** [comment]
The default should be dealt with earlier, meaning when setting up `optionst`.

**Review by martin-cs — CHANGES_REQUESTED** [review]
Daniel is correct; I should not have approved this.  The correct fix is an `else` here:

https://github.com/diffblue/cbmc/blob/06c563a76589ed00cc372d68e2819880537977b2/src/cbmc/cbmc_parse_options.cpp#L261

which sets the correct value.  It might be worth seeing if `unwind` has similar behaviour.

**Comment by TGWDB** [comment]
I tried changing the creation of the `optiont` and this now breaks unit tests (these can be fixed). I'm unclear why the `"depth"` option is a special case that should be handled outside, when other options passed in to the `symex_config` that are missing are handled in the constructor. (The main difference appears to be that `"depth"` has a different default value, but this is also handled in the constructor for the `"max_field_sensitivity_array_size"` option.)

**Comment by martin-cs** [comment]
@TGWDB It is a fair question.  In an ideal world the `*_parse_options.cpp` file should read the `cmdlinet` and use it to populate an `optionst` along with default values if they are not 0 (or whatever `optionst` returns for the "asking for an option that hasn't been set" case).  Unit tests can then simply create an appropriate `optionst`.  However it is made, it is the `optionst` which gets passed to the actual code and used to configure things.  Ideally this will be in a constructor, RAII-style.  There is a little bit of a grey area about whether and how the constructors should handle cases when the configuration is complex and has multiple, non-orthogonal options.

Historically, this bit of design intent has not been as widely followed as maybe it should have been, so in the older parts of the code you do find configuration parameters lurking much deeper.  These should be treated as a legacy issue and ideally modernised and moved into the "`parse_options.cpp` builds an `optionst` and that is it for configuration" world.

HTH

**Comment by TGWDB** [comment]
Rather than an explicit `else` case I set the default value for the option in the creation of `optionst`. This seems like the cleanest solution.

**Review by martin-cs — APPROVED** [review]
Seems a reasonable solution.

**Comment by TGWDB** [comment]
@kroening Any chance of a codeowner review now?

**Review by tautschnig — APPROVED** [review]

**Comment by tautschnig** [comment]
@TGWDB Please cleanup the commit history on this one.

**Comment by TGWDB** [comment]
> @TGWDB Please cleanup the commit history on this one.

Commit messages tidied up. This should just be waiting on codeowner review now.

**Comment by martin-cs** [comment]
Paging @peterschrammel

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**
